testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -1339,6 +1339,34 @@ public void testDeleteWholePartition()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testDeleteFromOriginalFiles()
+    {
+        withTemporaryTable("delete_original_files", true, true, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s WITH (transactional = true, partitioned_by = ARRAY['regionkey'])" +
+                    " AS SELECT nationkey, name, regionkey FROM tpch.tiny.nation", tableName));
+            // SELECT before deletion may prime the cache on the Hive side
+            verifySelectForPrestoAndHive("SELECT count(*) FROM " + tableName, "true", row(25));
```

Comment:
It would be nice to have a sanity check that we actually got a table with original files here as a result of Trino's CTAS to transactional table. Currently calling CTAS will create such table but it may change in the future.

Reply:
I updated the commit to add a method to verify that the data file paths for original files conform to the formats generated by either Hive or Trino, and call that method in all the tests of original files.  I also added a test of UPDATE of an unpartitioned original files table.